src/argo.cpp

```diff
@@ -22,9 +22,13 @@ mem::dynamic_memory_pool<alloc::global_allocator, mem::NODE_ZERO_ONLY> collectiv
 mem::dynamic_memory_pool<alloc::global_allocator, mem::ALWAYS> dynamic_prepool(&alloc::default_global_allocator);
 
 namespace argo {
-	void init(size_t size) {
+	void init(std::size_t size) {
 		vm::init();
-		default_global_mempool = new mem::global_memory_pool<>(size);
+		std::size_t requested_size = size;
+		using mp = mem::global_memory_pool<>;
+		requested_size += mp::reserved;
```

Comment:
This code is lifted as-is from the former constructor of `global_memory_pool`. It is indeed unclear why we do this, and I am not sure we should, however, I am also unsure how to correctly document this. Any ideas?

Reply:
Of course, the details of the documentation for it could depend on what the reserved space is actually used for. Without going into that, my general suggestion for documentation would be to (i) note in the documentation for the `global_memory_pool` class that `reserved` bytes are cornered of for internal use out of the backend size and (ii) have a comment on/near the line increasing the `requested_size` that it is added space for internal use in the memory pool.

Reply:
To deal with this I opened issue #22